pkg/streamingpromql/types/hpoint_ring_buffer.go

```diff
@@ -78,6 +78,8 @@ func (b *HPointRingBuffer) UnsafePoints(maxT int64) (head []promql.HPoint, tail
 // PutHPointSlice when it is no longer needed.
 // Calling UnsafePoints is more efficient than calling CopyPoints, as CopyPoints will create a new slice and copy all
 // points into the slice, whereas UnsafePoints returns a view into the internal state of this buffer.
+// In addition to copying the HPoint's, CopyPoints will also duplicate the FloatHistogram values
```

Comment:
[nit] We use the word "points" earlier, so this would keep it consistent

```suggestion
// In addition to copying the points, CopyPoints will also duplicate the FloatHistogram values
```